test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetRestoreTests.cs

```diff
@@ -3178,6 +3178,39 @@ public void DotnetRestore_WritesExpectedAuditModeInAssetsFile(string userAuditMo
             assetsFile.PackageSpec.RestoreMetadata.RestoreAuditProperties.AuditMode.Should().Be(expectedAuditMode);
         }
 
+        [InlineData("9.0.100", "warning NU1604")]
+        [InlineData("10.0.100", "error NU1015")]
+        [Theory]
+        public void DotnetRestore_PackageReferenceWithNoVersion_OutputExpectedDiagnostic(string sdkAnalysisLevel, string expected)
```

Comment:
You can probably easily rewrite this as a test that works without a second process: 

https://github.com/NuGet/NuGet.Client/blob/58c248e36695c9586c486e8876a43046e0817d41/test/NuGet.Core.FuncTests/NuGet.Commands.FuncTest/RestoreCommand_PrunePackageReference.cs#L140-L205

Reply:
I forgot about that being an issue so that's why the test is added.